src/contract/assignments.rs

```diff
@@ -31,6 +31,9 @@ use super::{
 };
 use crate::schema;
 
+//TODO: Find a better place for MergeReveal
+use crate::contract::MergeRevealed;
```

Comment:
I think this is stash-specific trait and we can move both trait and its implementation to a dedicated file in `src/stash/merge.rs`.

Also single-method traits must be named the same as the method itself - and I think that the better name is `MergeRevealed::merge_revealed()` (without `_with` suffix, which is usually used for constructors).

Reply:
Noted.